docs/grammar/SolidityLexer.g4

```diff
@@ -0,0 +1,335 @@
+Ufixed: 'ufixed' | ('ufixed' [0-9]+ 'x' [0-9]+);
+/**
+ * Sized unsigned integer types.
+ * uint is an alias of uint256.
+ */
+UnsignedIntegerType:
+	'uint' | 'uint8' | 'uint16' | 'uint24' | 'uint32' | 'uint40' | 'uint48' | 'uint56' | 'uint64' |
+	'uint72' | 'uint80' | 'uint88' | 'uint96' | 'uint104' | 'uint112' | 'uint120' | 'uint128' |
+	'uint136' | 'uint144' | 'uint152' | 'uint160' | 'uint168' | 'uint176' | 'uint184' | 'uint192' |
+	'uint200' | 'uint208' | 'uint216' | 'uint224' | 'uint232' | 'uint240' | 'uint248' | 'uint256';
+Using: 'using';
+View: 'view';
+Virtual: 'virtual';
+While: 'while';
+
+LParen: '(';
+RParen: ')';
+LBrack: '[';
+RBrack: ']';
+LBrace: '{';
+RBrace: '}';
+Colon: ':';
+Semicolon: ';';
+Period: '.';
+Conditional: '?';
+Arrow: '=>';
+
+Assign: '=';
+AssignBitOr: '|=';
+AssignBitXor: '^=';
+AssignBitAnd: '&=';
+AssignShl: '<<=';
+AssignSar: '>>=';
+AssignShr: '>>>=';
+AssignAdd: '+=';
+AssignSub: '-=';
+AssignMul: '*=';
+AssignDiv: '/=';
+AssignMod: '%=';
+
+Comma: ',';
+Or: '||';
+And: '&&';
+BitOr: '|';
+BitXor: '^';
+BitAnd: '&';
+Shl: '<<';
+Sar: '>>';
+Shr: '>>>';
+Add: '+';
+Sub: '-';
+Mul: '*';
+Div: '/';
+Mod: '%';
+Exp: '**';
+
+Equal: '==';
+NotEqual: '!=';
+LessThan: '<';
+GreaterThan: '>';
+LessThanOrEqual: '<=';
+GreaterThanOrEqual: '>=';
+Not: '!';
+BitNot: '~';
+Inc: '++';
+Dec: '--';
+
+/**
+ * A single quoted string literal restricted to printable characters.
+ */
+StringLiteral: '"' DoubleQuotedStringCharacter* '"' | '\'' SingleQuotedStringCharacter* '\'';
+/**
+ * A single non-empty quoted string literal.
+ */
+NonEmptyStringLiteral: '"' DoubleQuotedStringCharacter+ '"' | '\'' SingleQuotedStringCharacter+ '\'';
+// Note that this will also be used for Yul string literals.
+//@doc:inline
+fragment DoubleQuotedStringCharacter: DoubleQuotedPrintable | EscapeSequence;
+// Note that this will also be used for Yul string literals.
+//@doc:inline
+fragment SingleQuotedStringCharacter: SingleQuotedPrintable | EscapeSequence;
+/**
+ * Any printable character except single quote or back slash.
+ */
+fragment SingleQuotedPrintable: [\u0020-\u0026\u0028-\u005B\u005D-\u007E];
+/**
+ * Any printable character except double quote or back slash.
+ */
+fragment DoubleQuotedPrintable: [\u0020-\u0021\u0023-\u005B\u005D-\u007E];
+/**
+  * Escape sequence.
+  * Apart from common single character escape sequences, line breaks can be escaped
+  * as well as four hex digit unicode escapes \\uXXXX and two digit hex escape sequences \\xXX are allowed.
+  */
+fragment EscapeSequence:
+	'\\' (
+		['"\\bfnrtv\n\r]
+		| 'u' HexCharacter HexCharacter HexCharacter HexCharacter
+		| 'x' HexCharacter HexCharacter
+	);
+/**
+ * A single quoted string literal allowing arbitrary unicode characters.
+ */
+UnicodeStringLiteral:
+	'unicode"' DoubleQuotedUnicodeStringCharacter* '"'
+	| 'unicode\'' SingleQuotedUnicodeStringCharacter* '\'';
+//@doc:inline
+fragment DoubleQuotedUnicodeStringCharacter: ~["\r\n\\] | EscapeSequence;
+//@doc:inline
```

Comment:
Highlighting breaks here - what is the purpose of `//@doc:inline`?

Reply:
I thought it breaks because of the quote in the line above?